src/trace/step-function-service.ts

```diff
@@ -76,7 +76,10 @@ export class StepFunctionContextService {
     if (typeof event !== "object") return;
 
     // Extract Payload if available (Legacy lambda parsing)
-    if (typeof event.Payload === "object") {
+    if (
+      typeof event.Payload === "object" &&
+      (typeof event.Payload._datadog === "object" || this.isValidContextObject(event.Payload))
```

Comment:
Nit: We could make this more by adding optional chaining, i.e.

`typeof event?.Payload === "object"`.

Unless we already know for sure that event is defined, then we don't have to do this